src/bpftrace.cpp

```diff
@@ -2099,23 +2134,44 @@ bool BPFtrace::is_aslr_enabled(int pid)
   return false;
 }
 
-std::string BPFtrace::resolve_usym(uintptr_t addr, int pid, bool show_offset, bool show_module)
+std::string BPFtrace::resolve_usym(uintptr_t addr,
+                                   int pid,
+                                   int probe_id,
+                                   bool show_offset,
+                                   bool show_module)
 {
   struct bcc_symbol usym;
   std::ostringstream symbol;
   void *psyms = nullptr;
-  struct bcc_symbol_option symopts;
-
-  memset(&symopts, 0, sizeof(symopts));
-  symopts.use_debug_file = 1;
-  symopts.check_debug_file_crc = 1;
-  symopts.use_symbol_type = BCC_SYM_ALL_TYPES;
 
   if (resolve_user_symbols_)
   {
-    if (cache_user_symbols_)
+    std::string pid_exe = get_pid_exe(pid);
+    if (pid_exe.empty() && probe_id != -1)
+    {
+      // sometimes program cannot be determined from PID, typically when the
+      // process does not exist anymore; in that case, try to get program name
+      // from probe
+      auto probe_full = resolve_probe(probe_id);
+      size_t start = probe_full.find(':') + 1;
+      size_t end = probe_full.find(':', start);
+      pid_exe = probe_full.substr(start, end - start);
```

Comment:
Is the caching just expected to fail when using wildcards in probes, e.g. "uprobe:/bin/gc*:*"

A comment explaining the expected behaviour would be useful for future developers

Reply:
Yes, since the program is not expanded, we don't know which executable the perf event came from if a wildcard is used. I'll add a comment explaining this.